### PR TITLE
Recommend actions when name can't be found

### PR DESCRIPTION
**Pull request by Lutzifer**
### Checklist
- [X] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [X] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [X] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [X] I've updated the documentation if necessary.

### Description
When running 'fastlane actions xy' without a hit, there now is an additional line at the end with similar actions, e.g. for 'fastlane actions xy':

> Available actions with matching name: hg_add_tag, hg_commit_version_bump, hg_ensure_clean_status, hg_push

### Motivation and Context
This makes it easier to find e.g. all "tag" or "git" related actions, providing a nice shortcut for the user. I like it easy.

### CODE REVIEW

File: fastlane/lib/fastlane/documentation/actions_list.rb [hunk 1]
```diff
@@ -73,7 +73,17 @@ def self.show_details(filter: nil)
       else
         puts "Couldn't find action for the given filter.".red
         puts "==========================================\n".red
+
         print_all # show all available actions instead
+
+        if !filter.nil? && filter.length > 1
+          candidates = []
+          all_actions(nil) do |a, action_name|
+            candidates << [action_name] if action_name.include? filter
```
Comment (mfurtak) [inline fastlane/lib/fastlane/documentation/actions_list.rb]: It feels strange to me that we're making an Array of Arrays each containing a String. Is this correct/necessary?

Comment (Lutzifer) [inline fastlane/lib/fastlane/documentation/actions_list.rb]: ah, that was a leftover from previous code, fixed it as it was no longer necessary.

File: fastlane/lib/fastlane/documentation/actions_list.rb [hunk 2]
```diff
@@ -79,7 +79,7 @@ def self.show_details(filter: nil)
         if !filter.nil? && filter.length > 1
           candidates = []
           all_actions(nil) do |a, action_name|
```
Comment (hjanuschka) [inline fastlane/lib/fastlane/documentation/actions_list.rb]: can we use something different than `a` here, i guess its meant to be the hash key?

**Comment by Lutzifer** [comment]
"a" was formerly known as action, but rubocop went all "bad cop" on it for shadowing the existing action variable 😉

Von meinem iPhone gesendet

**Comment by hjanuschka** [comment]
@Lutzifer it seems to be not the action name but the ref to the instance, how about changing `a` to `action_ref`? only `a` is just too confusing

**Comment by hjanuschka** [comment]
@Lutzifer ruby >= 2.3 ships with the "DidYouMean" gem.


```ruby
a = DidYouMean::SpellChecker.new(dictionary: ['action1', 'action2'])
corrections =  a.correct("acton")
puts "Did you mean: #{corrections.join(',')}"
```

results in:
`Did you mean: action2,action1`


i think following paths are available:
  *  detect ruby >2.3
  * add https://github.com/yuki24/did_you_mean/ as dependency
    * it seems to be MIT


what do you guys think?

**Comment by Lutzifer** [comment]
That looks nice. The public interface in your example is new to v1.1.0 though (https://github.com/yuki24/did_you_mean/releases/tag/v1.1.0) and needs at least ruby 2.4.0

**Comment by KrauseFx** [comment]
`did_you_mean` is included in the default Ruby installation now?

**Comment by hjanuschka** [comment]
> Ruby 2.3 and later ships with this gem and it will automatically be required when a Ruby process starts up. No special setup is required.


however my sample uses 2.4 version (got it from their repo), so we'd need to find a way to do it with the 2.3 interface.

**Comment by Lutzifer** [comment]
as this is currently working for all versions and has additional recommendations on 2.4: can we get it merged or is the different behaviour a no-go?

**Comment by Lutzifer** [comment]
todo: add recommendations for "fastlane run" command

File: fastlane/lib/fastlane/documentation/actions_list.rb [hunk 3]
```diff
@@ -73,7 +73,29 @@ def self.show_details(filter: nil)
...
+    def self.print_suggestions(filter)
+      if !filter.nil? && filter.length > 1
+        action_names = []
+        all_actions(nil) do |action_ref, action_name|
+          action_names << action_name
+        end
+
+        corrections = []
+
+        if !Gem.loaded_specs["did_you_mean"].nil? && Gem.loaded_specs["did_you_mean"].version >= Gem::Version.new('1.1.0')
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/documentation/actions_list.rb]: Do we need `1.1.0`? I have `1.0.0` on my current Ruby installation.

Comment (KrauseFx) [inline fastlane/lib/fastlane/documentation/actions_list.rb]: Ok, seems like it doesn't work with `1.0.0`

```
bundler: failed to load command: fastlane (/Users/fkrause/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/bin/fastlane)
NameError: [!] uninitialized constant DidYouMean::SpellChecker
```

**Review by KrauseFx — COMMENTED** [review]
Thanks for working this, this change looks awesome, I'll take over and finish some smaller things up so we can merge it 👍

**Review by KrauseFx — APPROVED** [review]
Just tried this, looks great, made one small change, it's ready now 👍  Thanks for your contribution

**Comment by KrauseFx** [comment]
Could you confirm the CLA with the Google bot?

**Comment by Lutzifer** [comment]
great. how do i do that exactly?

**Comment by KrauseFx** [comment]
@Lutzifer Just say that you're ok with the CLA agreement and that this PR can be merged as a comment. This is just because the bot is confused as the PR is by multiple authors 👍

**Comment by Lutzifer** [comment]
I am ok with the cla and the pr can be merged!

**Comment by KrauseFx** [comment]
Beautiful, thanks 👍

**Pull request merged**